agent/exec/container/controller.go

```diff
@@ -23,6 +24,11 @@ type controller struct {
 	adapter *containerAdapter
 	closed  chan struct{}
 	err     error
+
+	pulled      chan struct{} // closed after pull
+	cancelPull  func()        // cancels pull context if not nil
+	pullErrOnce sync.Once
```

Comment:
Same comment as in the docker engine version: `pullErrOnce` shouldn't be necessary.